### PR TITLE
Add a test suite for the Image class

### PR DESCRIPTION
**Pull request by OverloadedOrama**
Part of #43440. I created a test suite for the Image class. This is my first time ever writing unit tests (and my second PR to Godot), so feedback is appreciated! The tests cover:

- [x] Image creation (create(), create_from_data(), copy_from())
- [x] Saving images to .png and .exr files.
- [x] Loading images from files (load() and load_x_from_buffer() methods, based on the work of @Calinou)
- [x] Getters, such as get_data(), get_size(), get_width(), get_height(), get_format(), get_used_rect(), get_rect(), get_pixel() and get_pixelv().
- [x] Image resizing with crop(), resize(), shrink_x2() and resize_to_po2().
- [x] Modifying the image's pixels with set_pixel(), set_pixelv(), fill(), blend_rect(), blit_rect(), flip_x() and flip_y().
- [x] Others, such as is_empty(), is_invisible(), is_compressed(), has_mipmaps().

The test suit is incomplete, the methods missing include most notably compression and mipmaps-related methods, bumpmap_to_normalmap(), srgb_to_linear() and some others, because I am not entirely sure how they work as I haven't used them so far.

### CODE REVIEW

File: tests/test_image.h [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+#define TEST_IMAGE_H
+
+#include "core/io/file_access_pack.h"
+#include "core/io/image.h"
+
+#include "thirdparty/doctest/doctest.h"
+
+namespace TestImage {
+
+TEST_CASE("[Image] Ways of creating images") {
```
Comment (Xrayez) [inline tests/test_image.h:41]: More concise and to the point.

```suggestion
TEST_CASE("[Image] Instantiation") {
```

File: tests/test_image.h [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+			err == OK,
+			"The image should load successfully using load().");
+	CHECK_MESSAGE(
+			image->get_data() == image_load->get_data(),
+			"The loaded image should have the same data as the one that got saved.");
+
+	const String base_dir = OS::get_singleton()->get_executable_path().get_base_dir();
+
+	// Load BMP
+	const String input_bmp_path = base_dir.plus_file("../tests/test_image/icon.bmp");
```
Comment (Xrayez) [inline tests/test_image.h:104]:
```suggestion
	// Load BMP
	const String input_bmp_path = TestUtils::get_data_path("icon.bmp");
```

To use `TestUtils::get_data_path`, you'll also need to include `#include "test_utils.h"`

But for this to work properly, you'll also need to move image files to `tests/data` directory which is created exactly for this. I think you can create a subfolder, like `tests/data/images` if you don't want to "clutter" the base test data directory. You'll then use relative path `TestUtils::get_data_path("images/icon.bmp")`.

You can also immediately open the BMP file without creating intermediate `input_bmp_path`:

```c++
FileAccessRef f_bmp = FileAccess::open(TestUtils::get_data_path("images/icon.bmp"), FileAccess::READ, &err);
```

The same applies to other paths.

File: tests/test_image.h [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+	CHECK_MESSAGE(
+			image_tga->load_tga_from_buffer(data_tga) == OK,
+			"The TGA image should load successfully.");
+}
+
+TEST_CASE("[Image] Basic getters") {
+	Ref<Image> image = memnew(Image(8, 4, false, Image::FORMAT_LA8));
+	CHECK_MESSAGE(
+			image->get_width() == 8,
+			"get_width() should return the expected value.");
```
Comment (Xrayez) [inline tests/test_image.h:163]: These type of messages are redundant in my opinion ("X expected to be Y"). I'd just use `CHECK()` macro for those. My rule of thumb is: if I start to write rationale which contains the "expected" word, I just replace `CHECK_MESSAGE` to `CHECK`.

File: tests/test_image.h (hunk 2 above)
Comment (OverloadedOrama) [reply tests/test_image.h]: Done. I didn't know about TestUtils, this is a much better way, thanks!

File: tests/test_image.h (hunk 3 above)
Comment (OverloadedOrama) [reply tests/test_image.h]: All right, I changed some of the checks to just `CHECK()`.

**Review by Xrayez — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**